core/src/mindustry/world/blocks/distribution/CraterConveyor.java

```diff
@@ -0,0 +1,271 @@
+            if(crater != null) crater.write(stream);
+
+            stream.writeInt(cooldown);
+        }
+
+        @Override
+        public void read(DataInput stream, byte revision) throws IOException{
+            super.read(stream, revision);
+
+            if(stream.readBoolean()) crater = new Crater(stream);
+
+            cooldown = stream.readInt();
+        }
+    }
+
+    protected class Crater implements Position{
+        float rotation;
+        float face;
+        float x;
+        float y;
+        Item item;
+
+        Crater(Tile tile){
+            x = tile.drawx();
+            y = tile.drawy();
+            rotation = tile.rotation() * 90 - 90;
+            face = rotation;
+        }
+
+        Crater(DataInput stream) throws IOException{
+            rotation = stream.readFloat();
+            face = stream.readFloat();
+            x = stream.readFloat();
+            y = stream.readFloat();
+        }
+
+        public void write(DataOutput stream) throws IOException{
+            stream.writeFloat(rotation);
+            stream.writeFloat(face);
+            stream.writeFloat(x);
+            stream.writeFloat(y);
+        }
+
+        @Override
+        public float getX(){
+            return x;
+        }
+
+        @Override
+        public float getY(){
+            return y;
+        }
+    }
+
+    @Override
+    public boolean acceptItem(Item item, Tile tile, Tile source){
+        CraterConveyorEntity entity = tile.ent();
+
+        if(!isStart(tile) && !(source.block() instanceof CraterConveyor)) return false;
+        if(entity.items.total() > 0 && !entity.items.has(item)) return false;
+        if(entity.items.total() >= getMaximumAccepted(tile, item)) return false;
+
+        return true;
+    }
+
+    @Override
+    public int getMaximumAccepted(Tile tile, Item item){
+        return Mathf.round(super.getMaximumAccepted(tile, item) * tile.entity.timeScale);
+    }
+
+    public boolean shouldLaunch(Tile tile){
+        CraterConveyorEntity entity = tile.ent();
+
+        // its not a start tile so it should be moving
+        if(!isStart(tile)) return true;
+
+        // its considered full
+        if(entity.items.total() >= getMaximumAccepted(tile, entity.crater.item)) return true;
+
+        return false;
+    }
+
+    @Override
+    public boolean blends(Tile tile, int rotation, int otherx, int othery, int otherrot, Block otherblock) {
+        return otherblock.outputsItems() && blendsArmored(tile, rotation, otherx, othery, otherrot, otherblock) && otherblock instanceof CraterConveyor;
+    }
+
+    // has no crater conveyors facing into it
+    private boolean isStart(Tile tile){
+        Tile[] inputs = new Tile[]{tile.back(), tile.left(), tile.right()};
+        for(Tile input : inputs){
+            if(input != null && input.getTeam() == tile.getTeam() && input.block() instanceof CraterConveyor && input.front() == tile) return false;
+        }
+
+        return true;
+    }
+
```

Comment:
these checks could/should probably interact with super's (undocumented) blendbits